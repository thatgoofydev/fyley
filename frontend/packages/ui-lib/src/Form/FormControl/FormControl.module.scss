.formControlWrapper {
  & + .formControlWrapper {
    margin-top: 10px;
  }
}