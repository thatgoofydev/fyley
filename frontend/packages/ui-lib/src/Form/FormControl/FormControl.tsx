import React, { FunctionComponent } from "react";
import styles from "./FormControl.module.scss";
import classNames from "classnames";

interface IFormControlProps {
  children: React.ReactNode;
  className?: string;
}

export const FormControl: FunctionComponent<IFormControlProps> = ({ className, children }) => {
  const classes = classNames(styles.formControlWrapper, className);
  return <div className={classes}>{children}</div>;
};
