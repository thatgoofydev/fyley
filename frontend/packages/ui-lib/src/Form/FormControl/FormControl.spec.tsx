import React from "react";
import { render, cleanup } from "@testing-library/react";
import { FormControl } from "./FormControl";

afterEach(cleanup);

describe("FormControl", function () {
  it("should render children", function () {
    const { container, getByText } = render(<FormControl>some text</FormControl>);

    expect(getByText("some text")).not.toBeNull();
    expect(container).toMatchSnapshot();
  });
});
