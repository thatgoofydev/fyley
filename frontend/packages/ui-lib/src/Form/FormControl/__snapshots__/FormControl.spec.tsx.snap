// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormControl should render children 1`] = `
<div>
  <div
    class="formControlWrapper"
  >
    some text
  </div>
</div>
`;
